content/src/debug.js

```diff
@@ -300,122 +302,54 @@ function showDebugMessage(m) {
-    return;
-  }
-  // Optimization: only compute line number when there is a visible
-  // async animation, OR, if it's one of the first 100 debug events.
-  if (record.line == null && record.exception &&
-      ((record.exited && record.appearCount > record.resolveCount) ||
-       record.debugId - firstSessionId < 100)) {
-    record.line = editorLineNumberForError(record.exception);
-  }
-  // TODO: schedule background computation of line numbers for debug
-  // events after the first 100.
-  if (record.line != null) {
-    var oldRecord = lineRecord[record.line];
-    if (record.appearCount > record.resolveCount) {
-      lineRecord[record.line] = record;
-      if (!oldRecord || !oldRecord.traced) {
-        traceLine(record.line);
-      }
-      record.traced = true;
-    } else {
-      if (!oldRecord || !oldRecord.appearCount || oldRecord === record) {
-        lineRecord[record.line] = record;
-        if (oldRecord && oldRecord.traced) {
-          untraceLine(record.line);
-        }
-      }
-      record.traced = false;
-    }
-  }
-  // Should we garbage-collect?  Here we do:
-  // The idea is this: we will only need to look up records by debugId
-  // in the future when there is a future event pending.  If all the
-  // expected "resolves" have occurred and all the expected "appears"
-  // have matching "resolves", and there has been an "exit" (matching
-  // the initial "enter", then there should be no further events for
-  // this debugId.
-  if (record.resolveCount >= record.totalCount &&
-      record.resolveCount >= record.appearCount &&
-      record.exited) {
-    delete debugIdRecord[record.debugId];
-  }
+  view.showProtractor(view.paneid('right'), step);
 }
 
-// Used while logging animations (during the 'appear' and 'resolve'
-// events) to grab information off an element that allows us to later
-// compute the coordinates.
-function collectCoords(elem) {
-  try {
-    // TODO: when the element is not a turtle with the standard
-    // parent element positioning, we should do a slower operation to
-    // grab the absolute position and direction.
-    return {
-      transform: elem.style[targetWindow.jQuery.support.transform]
-    };
-  } catch (e) {
-    return null;
+// The canonical 2D transforms written by this plugin have the form:
+// translate(tx, ty) rotate(rot) scale(sx, sy) rotate(twi)
+// (with each component optional).
+// This function quickly parses this form into a canonicalized object.
+function parseTurtleTransform(transform) {
+  if (transform === 'none') {
+    return {tx: 0, ty: 0, rot: 0, sx: 1, sy: 1, twi: 0};
   }
+  // Note that although the CSS spec doesn't allow 'e' in numbers, IE10
+  // and FF put them in there; so allow them.
+  var e = /^(?:translate\(([\-+.\de]+)(?:px)?,\s*([\-+.\de]+)(?:px)?\)\s*)?(?:rotate\(([\-+.\de]+)(?:deg)?\)\s*)?(?:scale\(([\-+.\de]+)(?:,\s*([\-+.\de]+))?\)\s*)?(?:rotate\(([\-+.\de]+)(?:deg)?\)\s*)?$/.exec(transform);
```

Comment:
Unfortunately Javascript doesn't have block regular expressions.

Perhaps a comment with an example that matches the pattern, for example

```js
// Matches CSS transforms of the form used by the turtle library, for example
// translate(10px, 20px) rotate(30deg) scale(2.0) rotate(-30deg)
```